python/src/aiconfig/eval/metrics.py

```diff
@@ -1,9 +1,27 @@
+from typing import Any, Generic
 
+from attr import dataclass
 from aiconfig.eval.common import (
     EvaluationMetricMetadata,
-    Metric,
+    SampleEvaluationFunction,
+    T_OutputDatum,
 )
 
+@dataclass(frozen=True)
+class Metric(Generic[T_OutputDatum]):
+    """See metrics.py for examples."""
+
+    evaluation_fn: SampleEvaluationFunction[T_OutputDatum]
```

Comment:
nit, should this just be `EvaluationFunction`?

Reply:
yep.